src/docs/manual-ad.rst

```diff
@@ -0,0 +1,304 @@
+    "(&(objectClass=user)(sAMAccountName=aduser))"
+
+After both ``kinit`` and ``ldapsearch`` work properly proceed to actual SSSD configuration.
+
+SSSD setup
+**********
+
+Configuring SSSD consists of several steps:
+
+* Install the ``sssd-ad`` package on the GNU/Linux client machine
+* Make configuration changes to the files below
+* Start the ``sssd`` service
+
+Copy the following sssd.conf, additional options can be added as needed
+
+.. code-block:: ini
+   :caption: /etc/sssd/sssd.conf
+
+    [sssd]
+    config_file_version = 2
+    domains = ad.example.com
+    services = nss, pam
+
+    [domain/ad.example.com]
+    # Uncomment if you need offline logins
+    # cache_credentials = true
+
+    id_provider = ad
+    auth_provider = ad
+    access_provider = ad
+
+    # Uncomment if service discovery is not working
+    # ad_server = server.ad.example.com
+
+    # Uncomment if you want to use POSIX UIDs and GIDs set on the AD side
+    # ldap_id_mapping = False
+
+    # Uncomment if the trusted domains are not reachable
+    #ad_enabled_domains = ad.example.com
+
+    # Comment out if the users have the shell and home dir set on the AD side
+    default_shell = /bin/bash
+    fallback_homedir = /home/%d/%u
+
+    # Uncomment and adjust if the default principal SHORTNAME$@REALM is not available
+    # ldap_sasl_authid = host/client.ad.example.com@AD.EXAMPLE.COM
+
+    # Comment out if you prefer to use shortnames.
+    use_fully_qualified_names = True
+
+    # Uncomment if the child domain is reachable, but only using a specific DC
+    # [domain/ad.example.com/child.example.com]
+    # ad_server = dc.child.example.com
+
+Set the file ownership and permissions
+
+.. code-block:: bash
+
+    chown root:root /etc/sssd/sssd.conf
+    chmod 0600 /etc/sssd/sssd.conf
+    restorecon /etc/sssd/sssd.conf
+
+NSS/PAM Configuration
+*********************
+
+Depending on your distribution you have different options how to enable SSSD.
+
+.. code-tabs::
+    :caption: Configure identity/authentication files
+
+    .. fedora-tab::
+
+        dnf install oddjob-mkhomedir
+        authselect select sssd with-mkhomedir
+        systemctl enable --now oddjobd.service
+
+    .. rhel-tab::
+
+        dnf install oddjob-mkhomedir
+        authselect select sssd with-mkhomedir
+        systemctl enable --now oddjobd.service
+
+    .. ubuntu-tab::
+
+        apt install libnss-sss libpam-sss
+
+On Debian/Ubuntu, add ``pam_mkhomedir.so`` to the PAM session configuration manually and restart SSSD.
+
+Configure NSS/PAM manually
+--------------------------
+
+Manual configuration can be done with the following changes. The file paths for PAM in the example below are from Debian/Ubuntu, in Fedora/RHEL corresponding manual configuration should be done in ``/etc/pam.d/system-auth`` and ``/etc/pam.d/password-auth``. See the sample
+nsswitch.conf below, it is expected to contain other modules.
+
+.. code-block:: bash
+   :caption: nsswitch
```

Comment:
Sorry, I did not mean to add nsswitch/pam as a caption but as a language instead of bash. I added those languages in latest commits to master, you will see them after rebase.

Reply:
Okay thanks, this is updated now.